extensions/incubator/src/main/java/io/opentelemetry/extension/incubator/trace/README.md

```diff
@@ -0,0 +1,167 @@
+
+// add propagationHeaders to request headers and call checkout service
+```
+
+<!-- markdownlint-disable -->
+```java
+// in checkout service: get request headers into a Map<String, String> requestHeaders
+Map<String, String> requestHeaders = new HashMap<>();
+String cartId = "cartId";
+
+SpanBuilder spanBuilder = tracer.spanBuilder("checkout_cart");
+String transactionId;
+
+TextMapGetter<Map<String, String>> TEXT_MAP_GETTER =
+    new TextMapGetter<Map<String, String>>() {
+      @Override
+      public Set<String> keys(Map<String, String> carrier) {
+        return carrier.keySet();
+      }
+
+      @Override
+      @Nullable
+      public String get(@Nullable Map<String, String> carrier, String key) {
+        //noinspection ConstantConditions
+        return carrier == null ? null : carrier.get(key);
+      }
+    };
+
+Map<String, String> normalizedTransport =
+    requestHeaders.entrySet().stream()
+        .collect(
+            Collectors.toMap(
+                entry -> entry.getKey().toLowerCase(Locale.ROOT), Map.Entry::getValue));
+Context newContext = openTelemetry
+    .getPropagators()
+    .getTextMapPropagator()
+    .extract(Context.current(), normalizedTransport, TEXT_MAP_GETTER);
+try (Scope ignore = newContext.makeCurrent()) {
+  Span span = spanBuilder.setSpanKind(SERVER).startSpan();
+  try (Scope scope = span.makeCurrent()) {
+    transactionId = processCheckout(cartId);
+  } catch (Throwable e) {
+    span.setStatus(StatusCode.ERROR);
+    span.recordException(e);
+    throw e; // or throw new RuntimeException(e) - depending on your error handling strategy
+  } finally {
+    span.end();
+  }
+}
+```
+<!-- markdownlint-enable -->
+
+After:
+
+```java
+Tracing tracing = new Tracing(openTelemetry, "service");
+Map<String, String> propagationHeaders = tracing.getTextMapPropagationContext();
+// add propagationHeaders to request headers and call checkout service
+```
+
+```java
+// in checkout service: get request headers into a Map<String, String> requestHeaders
+Map<String, String> requestHeaders = new HashMap<>();
+String cartId = "cartId";
+
+Tracing tracing = new Tracing(openTelemetry, "service");
+String transactionId = tracing.traceServerSpan(requestHeaders,
+    tracer.spanBuilder("checkout_cart"), () -> processCheckout(cartId));
+```
+
+Note:
+
+- You can use `traceConsumerSpan` if you want to trace a consumer
+  (e.g. from a message queue) instead of a server.
+- Exceptions are re-thrown without modification - see [Exception handling](#exception-handling)
+  for more details.
+
+### Setting baggage entries
+
+Before:
+
+```java
+BaggageBuilder builder = Baggage.current().toBuilder();
+builder.put("key", "value");
+Context context = builder.build().storeInContext(Context.current());
+try (Scope ignore = context.makeCurrent()) {
+  String value = Baggage.current().getEntryValue("key");
+}
+```
+
+After:
+
+```java
+Tracing tracing = new Tracing(openTelemetry, "service");
+String value = Tracing.callWithBaggage(
```

Comment:
Looks like these examples need to be updated to reflect the current naming. Let's hold off for now until we nail down the API, but remember to update before we merge.

Reply:
updated now